src/components/Search/index.js

```diff
@@ -1,16 +1,88 @@
-import React from 'react';
-import { Input } from '@material-ui/core';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { fade, makeStyles } from '@material-ui/core/styles';
+import SearchIcon from '@material-ui/icons/Search';
+import InputBase from '@material-ui/core/InputBase';
+
+const useStyles = makeStyles(theme => ({
+  search: {
+    position: 'relative',
+    borderRadius: theme.shape.borderRadius,
+    backgroundColor: fade(theme.palette.common.white, 0.15),
+    '&:hover': {
+      backgroundColor: fade(theme.palette.common.white, 0.25),
+    },
+    marginLeft: 0,
+    width: '100%',
+    [theme.breakpoints.up('sm')]: {
+      marginLeft: 0,
+      width: 'auto',
+    },
+  },
+  searchIcon: {
+    padding: theme.spacing(0, 2),
+    height: '100%',
+    position: 'absolute',
+    pointerEvents: 'none',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  inputRoot: {
+    color: 'inherit',
+  },
+  inputInput: {
+    border: '1px solid #a0a0a0',
+    padding: theme.spacing(1, 1, 1, 0),
+    // vertical padding + font size from searchIcon
+    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
+    transition: theme.transitions.create('width'),
+    width: '100%',
+    [theme.breakpoints.up('sm')]: {
+      width: '20ch',
+      '&:focus': {
+        width: '20ch',
+      },
+    },
+  },
+}));
+
+const Search = ({ label, search }) => {
+  const classes = useStyles();
+  const [searchValue, setSearchValue] = useState('');
+
+  const handleSearchInputChanges = e => {
+    setSearchValue(e.target.value);
+  };
+
+  const callSearchFunction = e => {
+    e.preventDefault();
+    search(searchValue);
+  };
 
-function Search({ label }) {
   return (
-    <div>
-      <Input type="search" placeholder={label} />
-    </div>
+    <form onSubmit={callSearchFunction} className={classes.search}>
+      <div className={classes.searchIcon}>
+        <SearchIcon />
+      </div>
+      <InputBase
```

Comment:
If we're going to have a very custom Input field here, it should probably be refactored out into it's own component, so that the design can be consistently used throughout the rest of the app.

Reply:
That's a great point!